.github/workflows/main.yml

```diff
@@ -60,8 +65,8 @@ jobs:
       with:
         token: ${{ secrets.CODECOV_TOKEN }} # not required for public repos
         file: |
-          ./coverage/coverage-final.json
-          ./coverage/jest/coverage-final.json
+          ./cypress-coverage/coverage-final.json
+          ./jest-coverage/coverage-final.json
         fail_ci_if_error: true # optional (default = false)
         path_to_write_report: ./coverage/codecov_report.txt
```

Comment:
@vickywane Looking at the codecov docs again, I think we can use this file to directly upload it as a combined coverage report artifact for github action and skip all the nyc merging code in this action workflow. Worth a try. Here is some example docs to look at:
https://github.com/codecov/codecov-action#example-workflowyml-with-codecov-action

Reply:
@vickywane Did you see this comment from 15 days ago? 
If you looked at the `out.json` file currently uploaded as a github artifact, you will notice that it is not convenient to read. Let's see if we can use a txt or html format that allows us to quickly understand the coverage report.